http-core/src/main/scala/org/apache/pekko/http/impl/engine/http2/hpack/ByteStringInputStream.scala

```diff
@@ -24,13 +27,29 @@ import pekko.util.ByteString.ByteString1C
 @InternalApi
 private[http2] object ByteStringInputStream {
 
-  def apply(bs: ByteString): InputStream =
-    bs match {
-      case cs: ByteString1C =>
-        // TODO optimise, ByteString needs to expose InputStream (esp if array backed, nice!)
-        new ByteArrayInputStream(cs.toArrayUnsafe())
-      case _ =>
-        // NOTE: We actually measured recently, and compact + use array was pretty good usually
-        apply(bs.compact)
+  private lazy val byteStringInputStreamMethodTypeOpt = Try {
+    val lookup = MethodHandles.publicLookup()
+    val inputStreamMethodType = MethodType.methodType(classOf[InputStream])
+    lookup.findVirtual(classOf[ByteString], "asInputStream", inputStreamMethodType)
+  }.toOption
+
+  def apply(bs: ByteString): InputStream = bs match {
+    case cs: ByteString1C =>
+      new ByteArrayInputStream(cs.toArrayUnsafe())
```

Comment:
Even the code is the same as the old, maybe a simple `instanceOf/asInstanceOf` will help the performance for this simple case.

Reply:
https://github.com/pjfanning/pekko-test/tree/instanceof -- this branch is slower than the main branch - the instanceof is slower than the pattern matching